pkg/scheduler/scheduler.go

```diff
@@ -235,8 +240,15 @@ func (s *ipClaimScheduler) autoAllocateExternalIP(svc *v1.Service, poolList *ext
 		glog.Errorf("Unable to update IP pool's '%v' allocation. Details: %v",
 			pool.Metadata.Name, err)
 	}
-
-	err = addServiceExternalIP(svc, s.Clientset, freeIP)
+	glog.V(5).Infof(
+		"Try to update externalIPs list of service '%v' with IP address '%v'",
+		svc.ObjectMeta.Name, freeIP,
+	)
+	svc.Spec.ExternalIPs = append(svc.Spec.ExternalIPs, freeIP)
+	if setLBIp {
+		svc.Spec.LoadBalancerIP = freeIP
```

Comment:
`svc.Spec.LoadBalancerIP` is used by LoadBalancer provider. You have to set in the status: https://github.com/lukaszo/bare-cloud-controller-manager/blob/lb/baremetal/loadbalancer.go#L188